server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java

```diff
@@ -2440,7 +2518,13 @@ public Map<String, Processor.Factory> getProcessors(Processor.Parameters paramet
             client,
             null,
             FailureStoreMetrics.NOOP,
-            TestProjectResolvers.alwaysThrow()
+            TestProjectResolvers.alwaysThrow(),
+            new FeatureService(List.of()) {
+                @Override
+                public boolean clusterHasFeature(ClusterState state, NodeFeature feature) {
+                    return DataStream.DATA_STREAM_FAILURE_STORE_FEATURE.equals(feature);
+                }
```

Comment:
It stands out as odd that this block of code is repeated over and over again. I don't have a great suggestion though -- maybe a static test helper FeatureService that just says yes to everything? That would avoid someone having to change this in 13 places if they add a new feature that they want used in all the tests.

Reply:
Yeah, `FeatureService` really feels like it wants a test implementation. I thought we had one but it seems to be unrelated. These were all to just get the tests happy, only one or two of them have any differentiating logic.